test/definitions/divorce/CaseTypeTab.test.js

```diff
@@ -0,0 +1,112 @@
+      const uniqResult = _.uniqWith(
+        caseTypeTab,
+        (field1, field2) =>
+          field1.TabID === field2.TabID &&
+          field1.CaseFieldID === field2.CaseFieldID
+      );
+      expect(uniqResult).to.eql(caseTypeTab);
+    });
+    it('should contain a unique tab field display order ID field tab ID (no duplicate field order in a tab)', function() {
+      const tabIds = _.uniq(_.map(caseTypeTab, 'TabID'));
+      tabIds.forEach(tabId => {
+        const allFieldsPerTab = _.filter(caseTypeTab, (field) => {
+          return field.TabID === tabId;
+        });
+        const uniqResults = _.uniqWith(
+          allFieldsPerTab,
+          (field1, field2) =>
+            field1.TabFieldDisplayOrder === field2.TabFieldDisplayOrder
+        );
+        expect(uniqResults).to.eql(allFieldsPerTab);
+      });
+    });
+  });
+
+  describe('sequence', function() {
+    it('should contain a proper sequence for TabFieldDisplayOrder with no gaps', function() {
+      const tabIds = _.uniq(_.map(caseTypeTab, 'TabID'));
+      tabIds.forEach(tabId => {
+        const allFieldsPerTab = _.filter(caseTypeTab, (field) => {
+          return field.TabID === tabId;
+        });
+        const allTabFieldDisplayOrderNumbers = _.map(allFieldsPerTab, (field) => {
+          return field.TabFieldDisplayOrder;
+        }).sort((a, b) => a - b);
+        for(var i = 1; i < allTabFieldDisplayOrderNumbers.length; i++) {
+          if(allTabFieldDisplayOrderNumbers[i] - allTabFieldDisplayOrderNumbers[i-1] !== 1) {
+            assert.fail(`Missing/unordered TabFieldDisplayOrder sequence number in TabID ${tabId} - expected ${allTabFieldDisplayOrderNumbers[i-1] + 1} but got ${allTabFieldDisplayOrderNumbers[i]}`);
+          }
+        }
+      });
+    });
+
+    const tabIds = _.uniq(_.map(caseTypeTab, 'TabID'));
+    const expected = {
+      History: 1,
+      petitionDetails: 2,
+      aosDetails: 3,
+      dnDetails: 4,
+      outcomeOfDnDetails: 5,
+      paymentDetails: 6,
+      documents: 7,
+      confidentialPetitioner: 8,
+      confidentialRespondent: 9,
+      confidentialCoRespondent: 10,
+      notes: 11,
+      marriageCertificate: 12,
+      coRespondent: 13,
+      SolicitorCoRespondent: 14,
+      LinkedCase: 15,
+      General: 1
+    };
+    tabIds.forEach(tabId => {
+      it(`all ${tabId} fields should have the expected tab order ${expected[tabId]}`, function() {
+
+        const allTabFields = _.uniq(_.filter(caseTypeTab, (field) => {
+          return field.TabID === tabId;
+        }));
+        const allTabOrders = _.uniq(_.map(allTabFields, 'TabDisplayOrder'));
+        expect(allTabOrders).to.eql([expected[tabId]]);
+      });
+    });
+  });
+
+  describe('validation', function() {
+    it('should contain a valid Tab IDs', function() {
+      const tabIds = _.uniq(_.map(caseTypeTab, 'TabID'));
+      expect(tabIds).to.eql([
+        "History",
```

Comment:
Mixing double-quotes with single quotes for strings - again es-lint might be good?

Reply:
Done 👍